stock_picking_report_internal_delivery_address/models/stock_picking.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2023 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def should_print_delivery_address(self):
```

Comment:
You should still call `super` on all the cases except the specific one we want to catch.